value.go

```diff
@@ -67,45 +68,213 @@ func getValue(info TypeInfo, v mapping.Value) (any, error) {
 	switch t {
+	case TYPE_SQLNULL:
+		return toPtr(mapping.CreateNullValue())
 	case TYPE_BOOLEAN:
-		vv, err = mapping.CreateBool(v.(bool)), nil
+		return toPtr(mapping.CreateBool(v.(bool)))
 	case TYPE_TINYINT:
-		vv, err = mapping.CreateInt8(v.(int8)), nil
+		return toPtr(mapping.CreateInt8(v.(int8)))
 	case TYPE_SMALLINT:
-		vv, err = mapping.CreateInt16(v.(int16)), nil
+		return toPtr(mapping.CreateInt16(v.(int16)))
 	case TYPE_INTEGER:
-		vv, err = mapping.CreateInt32(v.(int32)), nil
+		return toPtr(mapping.CreateInt32(v.(int32)))
 	case TYPE_BIGINT:
-		vv, err = mapping.CreateInt64(v.(int64)), nil
+		return toPtr(mapping.CreateInt64(v.(int64)))
 	case TYPE_UTINYINT:
-		vv, err = mapping.CreateUInt8(v.(uint8)), nil
+		return toPtr(mapping.CreateUInt8(v.(uint8)))
 	case TYPE_USMALLINT:
-		vv, err = mapping.CreateUInt16(v.(uint16)), nil
+		return toPtr(mapping.CreateUInt16(v.(uint16)))
 	case TYPE_UINTEGER:
-		vv, err = mapping.CreateUInt32(v.(uint32)), nil
+		return toPtr(mapping.CreateUInt32(v.(uint32)))
 	case TYPE_UBIGINT:
-		vv, err = mapping.CreateUInt64(v.(uint64)), nil
+		return toPtr(mapping.CreateUInt64(v.(uint64)))
 	case TYPE_FLOAT:
-		vv, err = mapping.CreateFloat(v.(float32)), nil
+		return toPtr(mapping.CreateFloat(v.(float32)))
 	case TYPE_DOUBLE:
-		vv, err = mapping.CreateDouble(v.(float64)), nil
+		return toPtr(mapping.CreateDouble(v.(float64)))
 	case TYPE_VARCHAR:
-		vv, err = mapping.CreateVarchar(v.(string)), nil
-	case TYPE_ARRAY:
-		return getMappedSliceValue(lt, t, v)
-	case TYPE_LIST:
-		return getMappedSliceValue(lt, t, v)
-	case TYPE_STRUCT:
-		return getMappedStructValue(lt, v)
+		return toPtr(mapping.CreateVarchar(v.(string)))
+	case TYPE_TIMESTAMP, TYPE_TIMESTAMP_TZ:
+		vv, err := getMappedTimestamp(v)
+		if err != nil {
+			return nil
+		}
+		return toPtr(mapping.CreateTimestamp(*vv))
+	case TYPE_TIMESTAMP_S:
+		vv, err := getMappedTimestampS(v)
+		if err != nil {
+			return nil
+		}
+		return toPtr(mapping.CreateTimestampS(*vv))
+	case TYPE_TIMESTAMP_MS:
+		vv, err := getMappedTimestampMS(v)
+		if err != nil {
+			return nil
+		}
+		return toPtr(mapping.CreateTimestampMS(*vv))
+	case TYPE_TIMESTAMP_NS:
+		vv, err := getMappedTimestampNS(v)
+		if err != nil {
+			return nil
+		}
+		return toPtr(mapping.CreateTimestampNS(*vv))
+	}
+	return nil
+}
+
+func toPtr[T any](x T) *T {
+	return &x
+}
+
+func getPointerValue(v any) any {
+	for {
+		if v == nil {
+			return nil
+		}
+		vo := reflect.ValueOf(v)
+		if vo.Kind() == reflect.Ptr {
+			if vo.IsNil() {
+				return nil
+			}
+			v = vo.Elem().Interface()
+			continue
+		}
+		return v
+	}
+}
+
+func isNil(i any) bool {
+	if i == nil {
+		return true
+	}
+
+	value := reflect.ValueOf(i)
+	kind := value.Kind()
+
+	switch kind {
+	case reflect.Chan, reflect.Func, reflect.Map, reflect.Ptr, reflect.Interface, reflect.Slice:
+		return value.IsNil()
+	}
+	return false
+}
+
+func createValueByReflection(v any) (Type, *mapping.Value, error) {
```

Comment:
We should return `typeInfo` to support nested types (lists in lists in lists, etc.). See my other comment below.